src/virtualenv/seed/embed/base_embed.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import absolute_import, unicode_literals
+
+from abc import ABCMeta
+
+import six
+
+from virtualenv.util.path import Path
+
+from ..seeder import Seeder
+
+
+@six.add_metaclass(ABCMeta)
+class BaseEmbed(Seeder):
+    def __init__(self, options):
+        super(BaseEmbed, self).__init__(options, enabled=options.without_pip is False)
+
+        self.download = options.download
+        self.extra_search_dir = [i.resolve() for i in options.extra_search_dir if i.exists()]
+        self.pip_version = None if options.pip == "latest" else options.pip
+        self.setuptools_version = None if options.setuptools == "latest" else options.setuptools
+        self.no_pip = options.no_pip
+        self.no_setuptools = options.no_setuptools
+
+    @classmethod
+    def add_parser_arguments(cls, parser):
+        group = parser.add_mutually_exclusive_group()
+        group.add_argument(
+            "--download",
+            dest="download",
+            action="store_true",
+            help="download latest pip/setuptools from PyPi",
+            default=False,
+        )
+        group.add_argument(
+            "--no-download",
+            "--never-download",
+            dest="download",
+            action="store_false",
+            help="do not download latest pip/setuptools from PyPi",
+            default=True,
+        )
+        parser.add_argument(
+            "--extra-search-dir",
+            metavar="d",
+            type=Path,
+            nargs="+",
+            help="a location containing wheels candidates to install from",
+            default=[],
+        )
+        for package in ["pip", "setuptools"]:
+            parser.add_argument(
+                "--{}".format(package),
+                dest=package,
+                metavar="version",
+                help="{} version to install, bundle for bundled".format(package),
+                default="latest",
+            )
+        for extra, package in [
+            ("", "pip"),
+            ("", "setuptools"),
+            ("N/A - kept only for backwards compatibility; ", "wheel"),
```

Comment:
I suspect the removal of `wheel` from virtualenvs will break more people than it should -- it was certainly an issue for us when wheel was introduced, I imagine the removal will be similarly annoying/painful

Reply:
The removal is there to mirror what venv does... venv must have had a good reason why to remove wheel 🤷‍♂ ?

Reply:
hmmmm maybe I'm misremembering the history here but I think `virtualenv` added `wheel` *after* `venv` existed -- and maybe it was just not added to `venv`?

Reply:
either way if venv can get away with not having it, and virtualenv now want to be feature parity with venv seems like a good idea to not have it 😊 unless you have use cases where not having wheel seeded breaks the world

Reply:
it makes default cache performance of `pip` way way worse (since it can't prebuild wheels and put them into cache) -- I'd rather we keep it

Reply:
@pfmoore as pip maintainer can you confirm that under venv build performance is significantly worse due to not bundling wheel? (if so I wonder why no one opened an upstream bug for it)

Reply:
for example:

```bash
#!/usr/bin/env bash
set -euxo pipefail

clean_env() {
    rm -rf ~/.cache/pip
    rm -rf venv
    virtualenv venv --no-download >& /dev/null
}

run_test() {
    venv/bin/pip install -qq pyyaml
    venv/bin/pip uninstall -qqy pyyaml
    time venv/bin/pip install -qq pyyaml
}


: with wheel
clean_env
run_test

: without wheel
clean_env
venv/bin/pip uninstall -qqy wheel
run_test
```


```console
$ bash t.sh 
+ : with wheel
+ clean_env
+ rm -rf /home/asottile/.cache/pip
+ rm -rf venv
+ virtualenv venv --no-download
+ run_test
+ venv/bin/pip install -qq pyyaml
+ venv/bin/pip uninstall -qqy pyyaml
+ venv/bin/pip install -qq pyyaml

real	0m0.900s
user	0m0.726s
sys	0m0.053s
+ : without wheel
+ clean_env
+ rm -rf /home/asottile/.cache/pip
+ rm -rf venv
+ virtualenv venv --no-download
+ venv/bin/pip uninstall -qqy wheel
+ run_test
+ venv/bin/pip install -qq pyyaml
+ venv/bin/pip uninstall -qqy pyyaml
+ venv/bin/pip install -qq pyyaml

real	0m9.808s
user	0m9.333s
sys	0m0.249s
```

without `wheel` installed, `pip` cannot build a wheel (and therefore can't cache the built output)

Reply:
oh, I'll add it back then, but puzzled how venv got away with not having it for the last 5 years 🤦‍♂

Reply:
addressed by https://github.com/pypa/virtualenv/issues/1486